src/MSBuild.UnitTests/CommandLineSwitches_Tests.cs

```diff
@@ -1329,6 +1330,37 @@ public void ProcessWarnAsMessageSwitchWithCodes()
             Assert.Equal(expectedWarningsAsMessages, actualWarningsAsMessages, StringComparer.OrdinalIgnoreCase);
         }
 
+        /// <summary>
+        /// Verifies that when the /profileevaluation switch is used with no values that an error is shown.
+        /// </summary>
+        [Fact]
+        public void ProcessProfileEvaluationEmpty()
+        {
+            CommandLineSwitches commandLineSwitches = new CommandLineSwitches();
+
+            MSBuildApp.GatherCommandLineSwitches(new ArrayList(new[] { "/profileevaluation" }), commandLineSwitches);
+
+            VerifySwitchError(commandLineSwitches, "/profileevaluation", AssemblyResources.GetString("MissingProfileParameterError"));
+        }
+
+        /// <summary>
+        /// Verifies that when the /profileevaluation switch is used with invalid filenames an error is shown.
+        /// </summary>
+        [InlineData(" ")]
+        [InlineData("a_file_with?invalid_chars")]
+        [InlineData("C:\\a_path\\with?invalid\\chars")]
```

Comment:
These aren't all invalid on other OSs which is why the tests are failing.

Reply:
Will make this test Windows-specific. Just trying to validate there is a validation going on around invalid chars, so any OS will do

---
In reply to: [154504123](https://github.com/Microsoft/msbuild/pull/2765#discussion_r154504123) [](ancestors = 154504123)

Reply:
In the end I made it slightly more generic to cover other OSs too

---
In reply to: [154737571](https://github.com/Microsoft/msbuild/pull/2765#discussion_r154737571) [](ancestors = 154737571,154504123)